### DISCUSSION TITLE
Offline logging

### DISCUSSION THREAD

**Discussion by Ben-Bock** [tbnobody/OpenDTU#176] [Ideas]
Hi, how about considering offline logging to a SD-Card as an option parallel to the MQTT (with same settings, e.g. appending all x seconds a new line to a csv)? There is cheap hardware available, which also includes a real time clock DS1307 so that the synced ntp time could be stored as well on the device (see  attached picture). 

Vielen Dank nochmals für das großartige Projekt!
![s-l500](https://user-images.githubusercontent.com/47773432/193532401-464f036b-57f0-4f10-8bda-6b88838774f8.jpg)

**Comment by ghost** [comment]
From my point of view a very interesting idea. I don't use MQTT and I don't plan to use it either - "graphic gimmicks" would then be possible with the data on the SD card, for example.
And: Great project, very stable operation, clear presentation.

**Comment by madmartin** [comment]
Interesting Idea!

About the board - I can only find similar boards:

https://www.berrybase.de/micro-sd-card-reader-modul-mit-spi-schnittstelle 

https://www.makershop.de/plattformen/arduino-shields/xd05-data-logger/

Can you give us a buying source example?

Another option: use a ESP32 CAM module  - it has a sd card slot too. I can test if the NRF24 module works on the HSPI connectors on GPIO12-15 on such a esp module.

**Reply by Ben-Bock** [reply]
The one I posted is chinaware, e.g. via e-bay ("mini data logger"). If you want to go for a trusted German source that also has good documentation, you may want to go for; https://www.reichelt.de/de/de/entwicklerboards-adalogger-featherwing-rtc-sd-slot-pcf8523-debo-feather-rtc-p235464.html?PROVID=2788&gclid=EAIaIQobChMImISlyPbI-gIVlYXVCh2HnQm9EAQYBCABEgIVNPD_BwE&&r=1. There are a lot of options out there.

**Comment by andy5macht** [comment]
"dokuhn" is working on new verion of his breakout board. this one includes a MicroSD port:
https://github.com/dokuhn/openDTU-BreakoutBoard
i would love to see support for offline logging - maybe even with a visualisation?! would be like magic for my eyes

**Comment by dontbyte** [comment]
FWIW: Ahoy implemented sd card storage last year:
https://github.com/stefan123t/ahoy/commit/60495de7cb7b05c966b48f7775e428962cf7f1cf
